src/main/drivers/bus_spi.c

```diff
@@ -124,170 +115,139 @@ uint32_t spiTimeoutUserCallback(SPI_TypeDef *instance)
 }
 
 
-FAST_CODE bool spiBusTransfer(const busDevice_t *bus, const uint8_t *txData, uint8_t *rxData, int length)
-{
-
-    #ifdef USE_DMA_SPI_DEVICE
-        if(USE_DMA_SPI_DEVICE == bus->busdev_u.spi.instance)
-        {
-            uint32_t timeoutCheck = millis();
-            memcpy(dmaTxBuffer, (uint8_t *)txData, length);
-            dmaSpiTransmitReceive(dmaTxBuffer, dmaRxBuffer, length, 1);
-            while(dmaSpiReadStatus != DMA_SPI_READ_DONE)
-            {
-                if(millis() - timeoutCheck > GYRO_READ_TIMEOUT)
-                {
-                    //GYRO_READ_TIMEOUT ms max, read failed, cleanup spi and return 0
-                    IOHi(bus->busdev_u.spi.csnPin);
-                    return false;
-                }
+FAST_CODE bool spiBusTransfer(const busDevice_t *bus, const uint8_t *txData, uint8_t *rxData, int length) {
+#ifdef USE_DMA_SPI_DEVICE
+    if(USE_DMA_SPI_DEVICE == bus->busdev_u.spi.instance) {
```

Comment:
Codacy found an issue: [Skipping configuration 'USE_DMA_SPI_DEVICE;USE_SPI' since the value of 'USE_DMA_SPI_DEVICE' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6307959)